rules/react.js

```diff
@@ -0,0 +1,63 @@
+"use strict";
+
+module.exports = {
+  parser: "babel-eslint",
+  plugins: ["react"],
+  rules: {
+    // enforces closing bracket after props always
+    "react/jsx-closing-bracket-location": [
+      "error",
+      {
+        selfClosing: "tag-aligned",
+        nonEmpty: "tag-aligned"
+      }
+    ],
+
+    // enforce spaces between curly braces
+    "react/jsx-curly-spacing": "error",
+
+    // do not enforce consistent props indentation
+    "react/jsx-indent-props": "error",
+
+    // prevent duplicate props in JSX
+    "react/jsx-no-duplicate-props": "error",
+
+    // disallow undeclared variables
+    "react/jsx-no-undef": "error",
+
+    // prevent React to be incorrectly marked as unused
+    "react/jsx-uses-react": "error",
+
+    // [revent variables used in JSX to be incorrectly marked as unused
+    "react/jsx-uses-vars": "error",
+
+    // prevent missing parentheses around multilines JSX
+    "react/jsx-wrap-multilines": "error",
+
+    // prevent usage of setState in componentDidMount
+    "react/no-did-mount-set-state": "error",
+
+    // prevent uage of setSate in componentDidUpdate
```

Comment:
nit: couple of missing letters here. "uage", setSate

Reply:
Just copied what was previously there, but I'll go in and fix it!